.github/workflows/build.yml

```diff
@@ -61,11 +61,11 @@ jobs:
         with:
           java-version: 11
 
-      - name: Upload Artifacts
-        run: ./gradlew uploadArchives
+      - name: Publish Artifacts
+        run: ./gradlew publishMavenPublicationToMavenCentralRepository paparazzi-gradle-plugin:publish
```

Comment:
Surprised this isn't just running 'publish' on everything. Why is such an elaborate command needed?

Reply:
Because layoutlib (and soon native libs) are pre-published ahead of paparazzi artifacts.  Also, publishMavenPublication doesn't seem to include the plugin marker.

Reply:
Works on licensee: https://repo1.maven.org/maven2/app/cash/licensee/

Reply:
`publish` is basically "publish all publications to all repositories (except maven local because Gradle doesn't like being consistent)". The next thing is `publishAllPublicationsTo[Repo]` so you're more specific on where to publish to but still publish everything. The most granular one is `publish[Publication]To[Repo]` which is what you are using.

Reply:
Licensee uses `publish` to "publish all the things": https://github.com/cashapp/licensee/blob/trunk/.github/workflows/build.yaml#L26

Paparazzi currently has a 2-phase deploy:
* phase 1 deploys the relocated layoutlib via the publishLayoutLibPublicationToMavenCentralRepository
* phase 2 deploys the paparazzi jvm artifacts via publishMavenPublicationToMavenCentralRepository

The explicit publications declaration for layoutlib [here](https://github.com/cashapp/paparazzi/blob/c4b951819128c2703ee73af3c44ec4c912655f6e/libs/layoutlib/build.gradle#L43) provides the `LayoutLibPublication` segment of the task name.

The implied publications declaration for the jvm artifacts provide the `MavenPublication` segment of the task name.

Why have 2 phases?  Cloning the layout-lib repo is rather slow, so currently the preference is to keep it out of band of CI builds.

Lastly, `publishMavenPublicationToMavenCentralRepository paparazzi-gradle-plugin:publish` is necessary, because the first task does not include/depend on the tasks to generate the plugin marker.

i.e., 

```
$ ./gradlew publishMavenPublicationToMavenLocal --console=plain

> Task :paparazzi-gradle-plugin:pluginVersion
> Task :paparazzi-gradle-plugin:compileKotlin
> Task :paparazzi-gradle-plugin:compileJava
> Task :paparazzi-gradle-plugin:pluginDescriptors
> Task :paparazzi-gradle-plugin:processResources
> Task :paparazzi-gradle-plugin:classes
> Task :paparazzi-gradle-plugin:inspectClassesForKotlinIC
> Task :paparazzi-gradle-plugin:jar
> Task :paparazzi-gradle-plugin:generateMetadataFileForMavenPublication
> Task :paparazzi-gradle-plugin:generatePomFileForMavenPublication
> Task :paparazzi-gradle-plugin:javaSourcesJar
> Task :paparazzi-gradle-plugin:javadoc
> Task :paparazzi-gradle-plugin:simpleJavadocJar
> Task :paparazzi-gradle-plugin:signMavenPublication
> Task :paparazzi-gradle-plugin:publishMavenPublicationToMavenLocal
```

vs.

```
$ ./gradlew paparazzi-gradle-plugin:publishToMavenLocal console=plain

> Task :paparazzi-gradle-plugin:pluginVersion
> Task :paparazzi-gradle-plugin:compileKotlin
> Task :paparazzi-gradle-plugin:compileJava
> Task :paparazzi-gradle-plugin:pluginDescriptors
> Task :paparazzi-gradle-plugin:processResources
> Task :paparazzi-gradle-plugin:classes
> Task :paparazzi-gradle-plugin:inspectClassesForKotlinIC
> Task :paparazzi-gradle-plugin:jar
> Task :paparazzi-gradle-plugin:generateMetadataFileForMavenPublication
> Task :paparazzi-gradle-plugin:generatePomFileForMavenPublication
> Task :paparazzi-gradle-plugin:javaSourcesJar
> Task :paparazzi-gradle-plugin:javadoc
> Task :paparazzi-gradle-plugin:simpleJavadocJar
> Task :paparazzi-gradle-plugin:signMavenPublication
> Task :paparazzi-gradle-plugin:publishMavenPublicationToMavenLocal
// additional tasks below
> Task :paparazzi-gradle-plugin:generatePomFileForPaparazziPluginMarkerMavenPublication
> Task :paparazzi-gradle-plugin:signPaparazziPluginMarkerMavenPublication
> Task :paparazzi-gradle-plugin:publishPaparazziPluginMarkerMavenPublicationToMavenLocal
> Task :paparazzi-gradle-plugin:generateMetadataFileForPluginMavenPublication
> Task :paparazzi-gradle-plugin:generatePomFileForPluginMavenPublication
> Task :paparazzi-gradle-plugin:signPluginMavenPublication
> Task :paparazzi-gradle-plugin:publishPluginMavenPublicationToMavenLocal
```